packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2893,11 +2888,6 @@ function commitRootImpl(
     const prevExecutionContext = executionContext;
     executionContext |= CommitContext;
 
-    // Reset this to null before calling lifecycles
-    if (__DEV__ || !disableStringRefs) {
-      setCurrentOwner(null);
-    }
```

Comment:
We weren't really initializing this to anything and we were always cleaning it up so I don't think this has been necessary for a while. However, since the life cycles also set the current debug fiber we get the resetting for free regardless - at least in DEV.